### ISSUE TITLE
Can't initialize GPS1 stream for  platformConfig I/O Preset 2

### ISSUE THREAD

**Issue by MorganMasters** [inertialsense/inertial-sense-sdk#735]
Hello, InertialSense Team,

I am working on getting a FLIR BlackFly camera (BFLY-PGE-23S6C-C) to trigger an InertialSense INS (RUG-3-IMX-5-Dual) inertial device on image capture, with the ultimate goal of integrating the triggering harness integrated into ROS2. Others in my group have done work on migrating your ROS wrapper to a ROS2 version.

I have successfully tested this harness outside of ROS2, using a modified version of `ExampleProjects/Communications/ISCommunicationsExample.cpp`. The modifications simply expose and enable `DID_STROBE_IN_TIME` messages.
My harness modified the MAIN cable as well, using lines 7, 9 as Tx, Rx (resp.) and line 2 as G9_STROBE. We chose to use I/O preset 2 (versus the default preset, 7) to accomplish this. For convenience, this means that I have TTL at S0, GPS1 at S1, and CAN on lines 11,12, instead of GPS2 on S0, RS232 on S1, and GPS1 on S2.

When I turned to the ROS2 integration, I found that my device initializes all streams of the `ros2/launch/example_params.yaml` except DID_GPS1_POS (see following stdout excerpt). Moreover, in EvalTool, I found that I/O Presets with GPS1 on a serial output other than S2 give a persistently flickering `GF[INIT GNSS1]` error on the info bar. I also notice that the driver fails to close its `/dev/ttyACM*` port, both for the ROS2 wrapper and the EvalTool, unless the I/O Preset is set to 7 (the default).

How, exactly, can I use non-default I/O Presets with ROS2 / can you explain to me where the configuration parameters are which allow me to initialize GPS1 on serial outputs which are not S2?

All the best!

``` 
mwmaster:~/ros2_ws/src/inertial-sense-sdk/ros2/launch$ ros2 run inertial_sense_ros2 new_target "example_params.yaml" 


Loading YAML paramfile: example_params.yaml

[INFO] [1729030461.473837842] [load_config_yaml]: InertialSenseROS: Loading configuration from YAML tree.
[INFO] [1729030461.474543168] [start]: ======  Starting Inertial Sense ROS2  ======
[INFO] [1729030461.474552176] [connect_to_serial]: InertialSenseROS: Connecting to serial port "/dev/ttyACM0", at 921600 baud
[/dev/ttyACM0] open():: Error opening port: 2
[ERROR] [1729030461.474567642] [open_port_error]: InertialSenseROS: Unable to open serial port "/dev/ttyACM0", at 921600 baud
[INFO] [1729030462.474800043] [connect_to_serial]: InertialSenseROS: Connecting to serial port "/dev/ttyACM1", at 921600 baud
[/dev/ttyACM1] open():: Error opening port: 2
[ERROR] [1729030462.474956448] [open_port_error]: InertialSenseROS: Unable to open serial port "/dev/ttyACM1", at 921600 baud
[INFO] [1729030463.475191864] [connect_to_serial]: InertialSenseROS: Connecting to serial port "/dev/ttyACM2", at 921600 baud
[INFO] [1729030466.190210563] [serial_port_connected_info]: InertialSenseROS: Connected to IMX SN60544 on "/dev/ttyACM2", at 921600 baud
[DEBUG] [1729030466.200947973] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030466.200988739] [flash_config]: InertialSenseROS: Attempting to enable flash config data stream
[DEBUG] [1729030466.201015382] [odom_ins_ned]: InertialSenseROS: Attempting to enable odom INS NED data stream
[DEBUG] [1729030466.201056287] [odom_ins_enu]: InertialSenseROS: Attempting to enable odom INS ENU data stream
[DEBUG] [1729030466.202159138] [covariance_stream]: InertialSenseROS: Attempting to enable DID_ROS_COVARIANCE_POSE_TWIST data stream
[DEBUG] [1729030466.202193729] [config_stream]: InertialSenseROS: Attempting to enable DID_INS_2 (5) data stream
[DEBUG] [1729030466.202233072] [config_stream]: InertialSenseROS: Attempting to enable DID_INL2_STATES (48) data stream
[DEBUG] [1729030466.202265121] [enable_nav_sat_fix_gps2]: InertialSenseROS: Attempting to enable gps2/NavSatFix
[DEBUG] [1729030466.202268361] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[DEBUG] [1729030466.202295854] [config_stream_gps_vel]: InertialSenseROS: Attempting to enable DID_GPS1_VEL (30) data stream
[DEBUG] [1729030466.202306871] [config_stream]: InertialSenseROS: Attempting to enable DID_GPS1_SAT (15) data stream
[DEBUG] [1729030466.202334932] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS2_POS (14) data stream
[DEBUG] [1729030466.203445065] [config_stream_gps_vel]: InertialSenseROS: Attempting to enable DID_GPS2_VEL (31) data stream
[DEBUG] [1729030466.203479751] [config_stream]: InertialSenseROS: Attempting to enable DID_MAGNETOMETER (52) data stream
[DEBUG] [1729030466.203519650] [config_stream]: InertialSenseROS: Attempting to enable DID_BAROMETER (53) data stream
[DEBUG] [1729030466.203561406] [config_stream]: InertialSenseROS: Attempting to enable DID_PIMU (3) data stream
[INFO] [1729030466.203613012] [config_rtk_rover]: InertialSenseROS: Configuring RTK Rover
[INFO] [1729030466.207318009] [successfully_connected_rtk]: InertialSenseROS: Successfully connected to RTK server [TCP:RTCM3:192.168.1.1:2101:mount0:user1:password1]. [Attempt 0]
[INFO] [1729030466.207490390] [set_rtkConfigBits]: InertialSenseROS: Setting rtkConfigBits: 0x00000002
[DEBUG] [1729030466.212396990] [stream_check]: DID_INS_4 response received
[INFO] [1729030466.212416704] [wait_for_reflla]: InertialSenseROS: Waiting for refLLA to be received from IMX
[INFO] [1729030466.212420062] [wait_for_reflla]: InertialSenseROS: Waiting for refLLA to be received from IMX
[DEBUG] [1729030466.212457938] [stream_check]: DID_MAGNETOMETER response received
[DEBUG] [1729030466.212488329] [stream_check]: DID_INL2_STATES response received
[DEBUG] [1729030466.213692280] [stream_check]: DID_INS_2 response received
[INFO] [1729030466.213700916] [wait_for_reflla]: InertialSenseROS: Waiting for refLLA to be received from IMX
[INFO] [1729030466.213704386] [wait_for_reflla]: InertialSenseROS: Waiting for refLLA to be received from IMX
[DEBUG] [1729030466.213708711] [stream_check]: DID_PIMU response received
[DEBUG] [1729030466.213730970] [stream_check]: DID_PIMU response received
[INFO] [1729030466.214875559] [wait_for_reflla]: InertialSenseROS: Waiting for refLLA to be received from IMX
[INFO] [1729030466.214884129] [wait_for_reflla]: InertialSenseROS: Waiting for refLLA to be received from IMX
[DEBUG] [1729030466.214906158] [stream_check]: DID_BAROMETER response received
[INFO] [1729030466.216484571] [wait_for_reflla]: InertialSenseROS: Waiting for refLLA to be received from IMX
[INFO] [1729030466.216493168] [wait_for_reflla]: InertialSenseROS: Waiting for refLLA to be received from IMX
[DEBUG] [1729030466.216505308] [IS_response_received]: InertialSenseROS: DID_FLASH_CONFIG response received
[DEBUG] [1729030466.216512119] [reflla_set]: InertialSenseROS: refLla was set
[DEBUG] [1729030466.415499959] [IS_response_received]: InertialSenseROS: DID_ROS_COVARIANCE_POSE_TWIST response received
[INFO] [1729030466.713551105] [diag_response_received]: InertialSenseROS: Diagnostics response received
[DEBUG] [1729030467.213673682] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030467.213726505] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[DEBUG] [1729030468.212681743] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030468.212738451] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[DEBUG] [1729030469.213534344] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030469.213583553] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[DEBUG] [1729030470.213501876] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030470.213549826] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[WARN] [1729030471.208464645] [rtk_correction]: Last received RTK correction data was 1.00088 seconds ago.
[DEBUG] [1729030471.213353135] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030471.213390021] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[DEBUG] [1729030472.213374266] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030472.213426144] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[DEBUG] [1729030473.213224731] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030473.213255804] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[DEBUG] [1729030474.213192415] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030474.213237901] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[WARN] [1729030475.208341833] [rtk_correction]: Last received RTK correction data was 0.99917 seconds ago.
[DEBUG] [1729030475.212378882] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030475.212421841] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[DEBUG] [1729030476.213220439] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030476.213262518] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[DEBUG] [1729030477.213038888] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030477.213066470] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[DEBUG] [1729030478.213023027] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030478.213067888] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[WARN] [1729030479.208139298] [rtk_correction]: Last received RTK correction data was 0.999926 seconds ago.
[DEBUG] [1729030479.213018196] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030479.213058143] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[DEBUG] [1729030480.212917278] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030480.212940678] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
[DEBUG] [1729030481.212916584] [gps1_pos]: InertialSenseROS: Attempting to enable GPS1 Pos data stream
[DEBUG] [1729030481.212948763] [config_stream_gps_pos]: InertialSenseROS: Attempting to enable DID_GPS1_POS (13) data stream
^C[INFO] [1729030481.350371234] [rclcpp]: signal_handler(signum=2)

```

**Comment by MorganMasters** [comment]
[ISCommunicationsExample.txt](https://github.com/user-attachments/files/17387633/ISCommunicationsExample.txt)
Modified sample code, resaved as a txt file.

**Comment by MorganMasters** [comment]
Resolved by upgrading from firmware 2.1.0 to 2.1.10 - according to Technical Support, there was a bug they fixed with respect to serial comms in recent firmware patches.
